.circleci/config.yml

```diff
@@ -272,6 +276,71 @@ jobs:
             docker login --username "${DOCKER_USER}" --password "${DOCKER_PASSWORD}"
             ./gradlew --no-daemon --parallel "-Pbranch=${CIRCLE_BRANCH}" dockerUpload
 
+  extractAPISpec:
+    executor: medium_executor
+    steps:
+      - prepare
+      - attach_workspace:
+          at: ~/project
+      - run:
+          name: Build runnable Teku binaries
+          command: |
+            ./gradlew --no-daemon --parallel installDist
+      - run:
+          name: Extract OpenAPI JSON
+          command: |
+            build/install/teku/bin/teku --network=mainnet --Xinterop-enabled=true --rest-api-docs-enabled=true &
+            TEKU_PID=$!
+            wget -O openapi.json http://localhost:5051/swagger-docs
```

Comment:
If wget fails because of a response error (404, 500, etc), this will still write `openapi.json`, but it won't contain what you want (it'll be empty, or it will contain the body of the error response). Also be aware that `store_artifacts` always runs, regardless if previous steps were successful. We likely want to do something like the following instead of the single `wget` line that's here now:

```bash
EXIT_CODE=0
wget -O openapi.json http://localhost:5051/swagger-docs
if [ $? != 0 ]; then
  rm openapi.json
  EXIT_CODE=1
fi

kill $TEKU_PID
exit $EXIT_CODE
```

This will cause the build to fail if the output isn't what we're expecting. There are probably more idiomatic ways to write this, however.

Reply:
added this and also added `--timeout=30` and  `--retry-connrefused` options to make sure it tries until service started (otherwise the command exits on connection refused directly) and to be sure of the timeout length (I did not see doc about the default value, so setting it at least indicates that we know there's a retry and timeout here and it's done on purpose)